docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,543 @@
+
+## Goals
+
+* Connecting and disconnecting UDNs and CUDNs within the same cluster
+  aka intra-cluster-udn-connectivity for admins (can be done as day0 or day2).
+  * Support for connecting `Layer3<->Layer3`, `Layer2<->Layer2` and `Layer3<->Layer2` type networks.
+  * Support for both `role:Primary` and `role:Secondary` CUDN connections.
+  * Support for both `role:Primary` and `role:Secondary` NAD connections. From
+    henceforth all references to CUDN/UDNs implies NADs as well.
+  * Support for both `role:Primary` UDN connections.
+* Support for pods, services (clusterIPs only since expectation today is that
+  NodePorts and LoadBalancer type services are already exposed externally and
+  hence reachable across UDNs) and network policies across the connected UDNs and CUDNs.
+* Ensure the proposed API is expandable for also connecting and disconnecting
+  UDNs or CUDNs across clusters in the future.
+
+## Future Goals
+
+* Tenants being able to request for interconnecting their networks
+  with other tenants.
+    * We need to understand exact use cases for why we need tenants to initiate
+      connection requests with other tenants. How will a tenant know other tenants
+      that exist on the cluster? In cases like finance tenant wanting to use IT
+      tenant's services, they could open a ticket to the admin to make that happen.
+* Admins being able to connect `role:Secondary` UDNs. Given
+  `SecondaryUserDefinedNetworkSelector` uses a combination of
+  namespace selector and network selector, a tenant could change
+  the labels on the UDN thereby causing security issues. Hence this
+  type of selector support will come later
+* Connecting and disconnecting UDNs and CUDNs across two clusters
+  aka inter-cluster-udn-connectivity for admins (can be done as day0 or day2).
+  There is a section that includes some basic details or open ended questions
+  at the end to continue this work for cross-clusters but that is not in scope
+  for the first phase of this enhancement. There are more basic designs that
+  need to be considered like "what does it mean for a UDN to exist across
+  two clusters" from an API standpoint and what is the multi-cluster
+  networking story for OVN-Kubernetes, before we can decide UDN connections
+  across clusters.
+
+## Non-Goals
+
+* Connecting two UDNs with overlapping subnets won’t be supported.
+  This is not a valid supported scenario. We did think about making
+  overlapping podIPs through services work, but even then it needs
+  some kind of SNATing because replies could still go to pods in
+  its own UDNs rather than to the other UDN.
+* Support for this feature in non-interconnect architecture is out of scope.
+* Support for `localnet` type network connections is out of scope.
+  * `localnet` type networks can be connected together using bridges already
+  * `localnet`<->`layer2` or `layer3` will not be covered here.
+* Supporting Live Migration and PersistentIPs across connected UDNs (Ask
```

Comment:
we should not support this imo

Reply:
since I have not heard back keeping this as non-goals